csharp/src/AddressBook/Addressbook.cs

```diff
@@ -62,7 +62,7 @@ public static partial class Addressbook {
   }
   #region Messages
   [global::System.Diagnostics.DebuggerNonUserCodeAttribute()]
-  public sealed partial class Person : pb::IMessage<Person>, global::System.IEquatable<Person> {
+  public sealed partial class Person : pb::IMessage<Person>, global::System.IEquatable<Person>, pb::IDeepCloneable<Person> {
```

Comment:
Yes, I'd be happy to make `IMessage<T>` extend `IDeepCloneable<T>` and `IEquatable<T>`, and `IFreezable` in the next PR. I do think it makes sense to have them as separate interfaces, but `IMessage<T>` can require them all.